File: leetcode/477.py
from typing import List


class Solution:

    def totalHammingDistance(self, nums: List[int]) -> int:
        ans, n = 0, len(nums)
        for i in range(32):
            c = 0
            for num in nums:
                c += (num >> i) & 1
            ans += c * (n - c)
        return ans

File: leetcode/477.c
int totalHammingDistance(int* nums, int numsSize) {
  int ans = 0;

  for (int i = 0; i < 32; ++i) {
    int count = 0;

    for (int j = 0; j < numsSize; ++j) {
      count += (nums[j] >> i) & 1;
    }

    ans += count * (numsSize - count);
  }

  return ans;
}